packages/block-library/src/template-part/index.js

```diff
@@ -50,3 +50,25 @@ addFilter(
 	'core/template-part',
 	enhanceTemplatePartVariations
 );
+
+// Prevent adding template parts inside post templates.
+addFilter(
+	'blockEditor.__unstableCanInsertBlockType',
+	'removeTemplatePartsFromPostTemplates',
+	(
+		can,
+		blockType,
+		rootClientId,
+		{ getBlock, getBlockParentsByBlockName }
+	) => {
+		if ( blockType.name !== 'core/template-part' ) {
+			return can;
+		}
+
+		const hasDisallowedParent =
+			getBlock( rootClientId )?.name === 'core/post-template' ||
```

Comment:
I guess this should be `core/post-content` here, no?

Reply:
I started that way, but then I thought _does it make sense to add template parts to post template?_ 

So let me ask you now – does it? :-)

Reply:
I guess it doesn't, but we should primarily check for `post-content`

Reply:
Fair point, I just added that.